subprojects/platform-play/src/main/java/org/gradle/play/internal/twirl/TwirlCompilerAdapterV10X.java

```diff
@@ -31,38 +31,15 @@
 import java.util.List;
 
 class TwirlCompilerAdapterV10X extends VersionedTwirlCompilerAdapter {
-    private static final Iterable<String> SHARED_PACKAGES = Arrays.asList("play.twirl.compiler", "scala.io"); //scala.io is for Codec which is a parameter to twirl
-
-    // Based on https://github.com/playframework/playframework/blob/2.4.0/framework/src/build-link/src/main/java/play/TemplateImports.java
-    private static final Collection<String> DEFAULT_JAVA_IMPORTS = Arrays.asList(
-        "models._",
-        "controllers._",
-        "play.api.templates.PlayMagic._",
-        "java.lang._",
-        "java.util._",
-        "scala.collection.JavaConversions._",
-        "scala.collection.JavaConverters._",
-        "play.api.i18n._",
-        "play.core.j.PlayMagicForJava._",
-        "play.mvc._",
-        "play.data._",
-        "play.api.data.Field",
-        "play.mvc.Http.Context.Implicit._");
-
-    private static final Collection<String> DEFAULT_SCALA_IMPORTS = Arrays.asList(
-        "models._",
-        "controllers._",
-        "play.api.templates.PlayMagic._",
-        "play.api.i18n._",
-        "play.api.mvc._",
-        "play.api.data._");
 
     private final String scalaVersion;
     private final String twirlVersion;
+    private final VersionedPlayTwirlAdapter playTwirlAdapter;
 
-    public TwirlCompilerAdapterV10X(String twirlVersion, String scalaVersion) {
+    public TwirlCompilerAdapterV10X(String twirlVersion, String scalaVersion, VersionedPlayTwirlAdapter playTwirlAdapter) {
```

Comment:
I think it would be good to have constructors like:

```java
public TwirlCompilerAdapterV10X(String scalaVersion, VersionedPlayTwirlAdapter playTwirlAdapter) {
    this("1.0.4", scalaVersion, playTwirlAdapter); // where "1.0.4" is the latest version of twirl for 1.0.x
}
```
This would make it easier to avoid mixing adapters and twirl version wrongly.